Review Simulation Control and General pages (#378)
lethe-cfd/lethe: doc/source/parameters/cfd/simulation_control.rst

* ``output control``: control for the output of the simulation results. The available options are: 
	* ``iteration``: results will be outputted at constant iteration frequency
	* ``time`` : results will be outputted at constant time

* ``output frequency``: controls after which number of iterations the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = iteration``.

.. tip::
	If ``set output frequency = 0``, no ``.pvd`` / ``.vtu`` file will be written.

	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the startup iteration will still be outputted.

* ``output time``: controls the time when the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = time``.

* ``group files``: number of ``.vtu`` files generated in a parallel simulation

.. tip::
	This parameter allows to reduce the number of files generated when the simulation is run with a large number of processors. ``set group files = 1`` ensures that a single ``.vtu`` file will be generated. In this case, the file is written using MPI IO functionalities.

> 	The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters.

Reviewer: ```suggestion
	The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters. However, as soon as the size of the output vtu file reaches 1Gb, it is preferable  to start splitting them into multiple smaller files as this may lead to corrupted files on some file systems.
```

Author: relevant indeed! Thank you

Simulation Control 
------------------

This subsection contains the general information of the simulation, including the time integration method and the general output names. It is the most commonly modified section for a simulation. 

.. tip::
	A standard convention in Lethe is to keep this section at the top of the parameter file, since it is generally the most accessed one.

.. seealso::
	For further understanding about the numerical method used and advanced parameters, the interested reader is referred to the Theory Documentation.

.. code-block:: text

  subsection simulation control

>     # The kind of solver for the linear system

Reviewer: This is not related to the linear system. It is the time stepping method.

    set method 	= steady

    #---------------------------------------------------
    # Steady-state simulation parameters
    #---------------------------------------------------
    # Number of mesh adaptation
    set number mesh adapt 	= 0

    # Tolerance at which the simulation is stopped
    set stop tolerance          = 1e-10

    #---------------------------------------------------
    # BDF scheme parameters
    #---------------------------------------------------
    # Method used to startup high order BDF methods
    set bdf startup method 	= multiple step bdf

>     # Scaling factor used in the iterations necessary to start-up the BDF schemes

Reviewer 2: ```suggestion
    # Scaling factor used in the iterations necessary to startup the BDF schemes
```

    set startup time scaling    = 0.4

    #---------------------------------------------------

>     # Time-stepping parameters

Reviewer: Transient simulations parameters

    #---------------------------------------------------
    # End time value of the simulation
    set time end        = 1
  
    # Time step value

>     set time step       = 1.

Reviewer 2: ```suggestion
    set time step       = 1
```

Reviewer 2: The default is "1.", but to be homogeneous I'd remove the "."

    # File output path
    set output path       = ./

    # File output prefix
    set output name                  = out
    set output name       = out

    # The control for the output of the simulation results
    set output control    = iteration
  
    # File output path
    set output path                  = ./
    # Output frequency
    set output frequency  = 1
  
    # Output time
    set output time                  = 1
  
    # Scaling factor used in the iterations necessary to start-up the BDF
    # schemes.
    set startup time scaling         = 0.4
  
    # Tolerance at which the simulation is stopped
    set stop tolerance               = 1e-10
  
    set output time       = 1

    # Maximal number of vtu output files
    set group files       = 1

    # Output the boundaries of the domain along with their ID
    set output boundaries = false
    
    # Subdivision of mesh cell in postprocessing
    set subdivision                  = 1
  
    # End time value of the simulation
    set time end                     = 1
  
    # Time step value
    set time step                    = 1.
    set subdivision       = 1

  end

> * The ``group files`` parameter controls the number of vtu files generated in a parallel simulation. This parameter allows to reduce the number of files generated when the simulation is run with a large number of processors. Setting group files to 1 ensures that there is a single vtu file generated. In this case, the file is written using MPI IO functionalities. The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters.

Reviewer: I prefer the previous version. This is a very subtly parameter and I rather have it more verbose than not verbose

Author: Don't worry, I kept all this information, this is of course relevant! I just put it in the tip-box just below, I think it is clearer when displayed in a web browser

> * ``number of mesh adapt``: number of mesh adaptation during the steady-sate simulation

Reviewer 2: ```suggestion
* ``number mesh adapt``: number of mesh adaptations during the steady-sate simulation
```

Author: eyes of the tiger! :tiger: ;)

> * ``stop tolerance``: tolerance at which the adjoint time stepping steady state simulations (``method = steady-bdf``) stops. 

Reviewer 2: ```suggestion
* ``stop tolerance``: tolerance at which the adjoint time stepping steady state simulation (``method = steady-bdf``) stops. 
```

> * ``bdf startup method``: scheme used to start a high order bdf scheme. The available options are: 

Reviewer 2: Maybe we should specify if it's two and above?

> * ``startup time scaling``: scaling factor used in the iterations necessary to start-up the BDF schemes

Reviewer 2: ```suggestion
* ``startup time scaling``: scaling factor used in the iterations necessary to startup the BDF schemes
```

	If ``set output frequency = 0``, no ``.pvd`` / ``.vtu`` file will be written.

> 	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the first (start-up) iteration will still be outputted.

Reviewer 2: ```suggestion
	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the first (startup) iteration will still be outputted.
```

* ``output time``: controls the time when the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = time``.

> * ``group files``: number of ``vtu`` files generated in a parallel simulation

Reviewer 2: ```suggestion
* ``group files``: number of ``.vtu`` files generated in a parallel simulation
```
